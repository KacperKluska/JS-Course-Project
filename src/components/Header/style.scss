@import '../../_colors';

#mainHeader {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  background-color: $primary-color;
  color: white;

  border-bottom: $secondary-color solid 1px;
  height: 80px;

  position: relative;
  z-index: 100;
}

#firstNavbar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 100%;
  min-height: 60px;
  width: 55%;
}

#leftNavDiv {
  display: flex;
  height: 100%;
  z-index: 100;
}

#outsideAlert {
  display: flex;
  align-items: center;

  width: 45%;

  z-index: 10;
}

.logo {
  font-size: 2.5rem;
  cursor: pointer;
  color: #352b2b;
}

#leftNav,
#rightNav {
  list-style: none;
  height: 100%;

  .navItemButton {
    padding: 1rem;
  }

  li {
    // padding: 1rem;
    display: flex;
    align-items: center;
  }

  li:hover {
    background-color: $secondary-color;
  }
}

#leftNav,
#rightNav {
  display: flex;
}

@media (max-width: 700px) {
  #outsideAlert {
    width: 0px;
  }

  #firstNavbar {
    width: 100%;
  }

  #leftNav {
    display: none;
    position: absolute;
    top: 0;
    flex-direction: column;
    width: 50vw;
    height: 100vh;
    background-color: $primary-color;
    z-index: 100;

    border-right: 1px solid $primary-color - #111;

    li {
      flex-direction: row;
      justify-content: center;
    }

    a {
      li {
        width: 100%;
      }
    }
  }

  #outsideAlert.active {
    position: absolute;
    top: 0;
  }

  #leftNav.active {
    display: flex;
    animation: showLeftMenu 0.5s ease-in-out;
  }

  #leftNav.disable {
    animation: hideLeftMenu 0.5s ease-in-out forwards;
    display: flex;
  }
}

@keyframes showLeftMenu {
  0% {
    transform: translateX(-100%);
  }
  100% {
    transform: translateX(0%);
  }
}

@keyframes hideLeftMenu {
  0% {
    transform: translateX(0%);
  }
  100% {
    transform: translateX(-100%);
  }
}

@media (max-width: 400px) {
  #rightNav {
    li {
      padding: 0.3rem;
    }
  }

  #logo {
    font-size: 1.8rem;
    cursor: pointer;
  }
}
