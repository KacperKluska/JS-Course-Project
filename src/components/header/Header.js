import NavButton from "../NavButton/NavButton";
import "./style.scss";
import OutsideAlerter from "../OutsideClick/OutsideAlerter";
import { useState } from "react";

export default function Header({ searchVisible, setSearchVisible }) {
  const [leftMenu, setLeftMenu] = useState(false);

  // const handleLeftMenu = () => {
  //   setLeftMenu((prev) => (prev = !prev));
  // };

  const handleLeftMenuShow = () => {
    setLeftMenu(true);
  };

  const handleLeftMenuClose = () => {
    setLeftMenu(false);
  };

  const handleSearch = () => {
    setSearchVisible();
    const searchInput = document.getElementById("searchInput");
    searchInput.focus();
  };

  return (
    <header id="mainHeader">
      <nav id="firstNavbar">
        <OutsideAlerter
          id="outsideAlert"
          func={handleLeftMenuClose}
          class={leftMenu ? "active" : null}
        >
          <div id="leftNavDiv">
            <ul id="leftNav" className={leftMenu ? "active" : "disable"}>
              <NavButton icon="icon-login">Sign in</NavButton>
              <NavButton>Register</NavButton>
              <NavButton
                icon="icon-search"
                onClick={() => {
                  handleSearch();
                  handleLeftMenuClose();
                }}
              >
                Search
              </NavButton>
            </ul>
          </div>
          <div
            id="leftNavPhone"
            onClick={handleLeftMenuShow}
            // style={{ display: leftMenu ? "none" : "flex" }}
            className={leftMenu ? "hide" : null}
          >
            <span className="bar"></span>
            <span className="bar"></span>
            <span className="bar"></span>
          </div>
        </OutsideAlerter>
        <h1 id="logo">SuitUp</h1>
        <ul id="rightNav">
          <NavButton icon="icon-gift">Gift Certificate</NavButton>
          <NavButton icon="icon-basket">Cart</NavButton>
        </ul>
      </nav>
    </header>
  );
}
