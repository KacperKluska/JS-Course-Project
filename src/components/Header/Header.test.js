import { BrowserRouter as Router } from "react-router-dom";
import Header from "./Header";

let wrapper;

beforeAll(() => {
  wrapper = mount(
    <Router>
      <Header />
    </Router>
  );
});

test("Header snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});
