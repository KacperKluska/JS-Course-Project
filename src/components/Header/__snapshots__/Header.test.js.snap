// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<header
  id="mainHeader"
>
  <div
    className=""
    id="outsideAlert"
  >
    <div
      id="leftNavDiv"
    >
      <ul
        className=""
        id="leftNav"
      >
        <a
          className="link"
          href="/login"
          onClick={[Function]}
        >
          <li
            className="navItem"
          >
            <i
              className="icon-login"
            />
            Sign in
          </li>
        </a>
        <a
          className="link"
          href="/register"
          onClick={[Function]}
        >
          <li
            className="navItem"
          >
            <i />
            Register
          </li>
        </a>
        <li
          className="navItem"
          onClick={[Function]}
        >
          <i
            className="icon-search"
          />
          Search
        </li>
      </ul>
    </div>
  </div>
  <nav
    id="firstNavbar"
  >
    <div
      className="hamburgerButton"
      onClick={[Function]}
    >
      <span
        className="bar"
      />
      <span
        className="bar"
      />
      <span
        className="bar"
      />
    </div>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      <h1
        id="logo"
      >
        SuitUp
      </h1>
    </a>
    <ul
      id="rightNav"
    >
      <li
        className="navItem"
      >
        <i
          className="icon-gift"
        />
        Gift Certificate
      </li>
      <li
        className="navItem"
      >
        <i
          className="icon-basket"
        />
        Cart
      </li>
    </ul>
  </nav>
</header>
`;
